factor/totp/classes/factor.php

```diff
@@ -46,37 +46,19 @@
 
 class factor extends object_factor_base {
 
-    public function verify($data) {
-        global $USER;
-        $factors = $this->get_user_factors($USER->id);
-
-        foreach ($factors as $factor) {
-            if ($factor->disabled == 0) {
-                $secret = $factor->secret;
-                $hotp = TOTP::create($secret);
-                $otp = $hotp->now();
-
-                if ($data['verificationcode'] !== $otp) {
-                    return array('verificationcode' => 'Wrong verification code');
-                }
-            }
-        }
-        return array();
-    }
-
     public function draw_qrcode($secretcode) {
         global $USER;
-        $code = 'otpauth://totp/'.$USER->username.':'.$USER->email.'?secret='.$secretcode.'&issuer=Moodle&algorithm=SHA1&&period=30';
+        $code = 'otpauth://totp/'.$USER->username.'_2:'.$USER->email.'?secret='.$secretcode.'&issuer=Moodle&algorithm=SHA1&&period=30';
```

Comment:
typo?

Reply:
In general, yes. Will discuss it later. Corrected